host/windows/syscall.c

```diff
@@ -480,142 +1130,525 @@ ssize_t oe_syscall_writev_ocall(
 
 oe_off_t oe_syscall_lseek_ocall(oe_host_fd_t fd, oe_off_t offset, int whence)
 {
-    OE_UNUSED(fd);
-    OE_UNUSED(offset);
-    OE_UNUSED(whence);
+    ssize_t ret = -1;
+    DWORD sfp_rtn = 0;
+    LARGE_INTEGER new_offset = {0};
 
-    PANIC;
+    new_offset.QuadPart = offset;
+    if (!SetFilePointerEx(
+            (HANDLE)fd, new_offset, (PLARGE_INTEGER)&new_offset, whence))
+    {
+        _set_errno(_winerr_to_errno(GetLastError()));
+        goto done;
+    }
+
+    ret = (oe_off_t)new_offset.QuadPart;
+
+done:
+    return ret;
 }
 
 int oe_syscall_close_ocall(oe_host_fd_t fd)
 {
-    return _close((int)fd);
+    int ret = -1;
+    HANDLE handle = (HANDLE)fd;
```

Comment:
Right, if a socket fd is passed in, this needs to call closesocket rather than CloseHandle.   Line 1387 does a check that tries to determine whether a handle is a socket handle or file handle, you might as well do the same check in this function around line 1343.

Reply:
Fixed with a helper function _is_valid_file_handle(), thanks.